nixos/modules/services/security/kanidm.nix

```
@@ -881,7 +881,14 @@ in
         (
           defaultServiceConfig
           // {
-            BindReadOnlyPaths = mergePaths (defaultServiceConfig.BindReadOnlyPaths ++ secretDirectories);
+            BindReadOnlyPaths = mergePaths (
+              defaultServiceConfig.BindReadOnlyPaths
+              ++ secretDirectories
+              ++ (lib.optionals (cfg.provision.enable && !cfg.provision.acceptInvalidCerts) [
+                "/etc/ssl"
+                "/etc/static/ssl"
```

Makes sense!
I think we should prefix the paths with `-` (`"-/etc/ssl"`) like it’s done for `kanidm-unixd`.
LGTM apart from that

---

I'd agree, but those paths are guarded with opt-in certificate validation. So the paths are required for validation, and If they fail to mount the provisioning script hangs and exits without logging to the journal. So my original argument against fail-closed was that you'll at least get a clear error message.

Thinking about it now; flexibility to define your own ca-bundle for verification and overwriting the path with ENV might be a valid use case, then fail closed should be used.
You're welcome to edit the PR, otherwise I'll update later today.